spec/requests/well_known/oauth_metadata_spec.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'The /.well-known/oauth-authorization-server request' do
+  let(:protocol) { ENV.fetch('LOCAL_HTTPS', true) ? :https : :http }
+
+  before do
+    host! ENV.fetch('LOCAL_DOMAIN')
+  end
```

Comment:
This feels non-idiomatic, and like I shouldn't need to be doing it here in this spec.

Reply:
See #29205

Reply:
This is not ideal, but it looks ok enough to me. I don't think that's a blocker.